test/unit/unitary2.cpp

```diff
@@ -0,0 +1,136 @@
+#include <gtest/gtest.h>
+#include "unitary/unitary.hpp"
+#include <string>
+
+#include <cstdio>
+#include <iostream>
+#include <memory>
+#include <stdexcept>
+#include <array>
+
+using namespace NRG::Unitary;
+using namespace NRG;
+using namespace std::string_literals;
+
+TEST(unitary, count_words_in_string){
+
+    EXPECT_EQ(count_words_in_string("one two three four a"s), 5);
+    EXPECT_EQ(count_words_in_string(""s), 0);
+
+}
+
+TEST(unitary, get_dims){
+
+    auto file = safe_open_for_reading("matrix.txt");
+    auto const [dim1, dim2] = get_dims(file);
+
+    EXPECT_EQ(dim1, 3);
+    EXPECT_EQ(dim2, 4);
+
+    auto file_err = safe_open_for_reading("matrix_err.txt");
+    EXPECT_THROW(get_dims(file_err), std::runtime_error);
+
+}
+
+
+void compare_matrices(MAT m1, MAT m2){
+    ASSERT_EQ(m1.size1(), m2.size1());
+    ASSERT_EQ(m1.size2(), m2.size2());
+    for(int i = 0; i < m1.size1(); i++){
+        for(int j = 0; j < m1.size2(); j++){
+            EXPECT_EQ(m1(i, j), m2(i, j));
+        }
+    }
+}
+
+template<typename T>
+void compare_matrices(vector<vector<T>> m1, MAT m2){
+    ASSERT_EQ(m2.size1(), m1.size());
+    ASSERT_EQ(m2.size2(), m1[0].size());
+    for(int i = 0; i < m2.size1(); i++){
+        for(int j = 0; j < m2.size2(); j++){
+            EXPECT_EQ(m2(i, j), m1[i][j]);
+        }
+    }
+    
+}
+
+
+TEST(unitary, read_matrix_text){
+
+    vector<vector<int>> const matrix_compare = {{31,41,53,46},{12,5,1,41},{5,2,4,7}};
+    auto matrix = read_matrix("matrix.txt");
+    compare_matrices(matrix_compare, matrix);
+}
+
+TEST(unitary, save_matrix){
+    auto matrix = read_matrix("matrix.txt");
+    save_matrix("matrix_temp.txt", matrix);
+    auto matrix_temp = read_matrix("matrix_temp.txt");
+    compare_matrices(matrix, matrix_temp);
+    std::remove("matrix_temp.txt");
+}
+
+
+std::string exec(const char* cmd) {
+    std::array<char, 128> buffer;
+    std::string result;
+    std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd, "r"), pclose);
+    if (!pipe) {
+        throw std::runtime_error("popen() failed!");
+    }
+    while (fgets(buffer.data(), buffer.size(), pipe.get()) != nullptr) {
+        result += buffer.data();
+    }
+    return result;
+}
+
+TEST(unitary, unitary_help){
+    
+    auto out = exec("../../tools/unitary -h");
```

Comment:
Relativne hard-coded poti so lahko problematicne. Verjetno bo tu potrebna pomoc CMake-a in v nek header zapisati absolutno pot do build direktorija z binary datoteka.